internal/node/hybrid/aws.go

```diff
@@ -43,7 +39,8 @@ func (hnp *HybridNodeProvider) ConfigureAws(ctx context.Context) error {
 			return fmt.Errorf("configuring aws credentials with IAM Roles Anywhere: %w", err)
 		}
 
-		awsConfig, err := LoadAWSConfigForRolesAnywhere(ctx, hnp.nodeConfig)
+		hnp.logger.Info("Waiting for AWS config to be available")
+		awsConfig, err := iamrolesanywhere.WaitForAWSConfig(configCtx, hnp.nodeConfig, 2*time.Second, iamrolesanywhere.EksHybridAwsCredentialsPath, hnp.logger)
```

Comment:
i know this follows the same pattern we had
but given the wait is conditional to using pod identity or not, i wonder if this wait should be part of the code that configures the `update` service. Like almost as a postflight validation: wait until the service you just started creates the creds file. it also allows us to print the error closer to where it actually happens